Can't find E0704 error code info

**Environment**: rustc 1.31.0-nightly (4bd4e4130 2018-10-25) and rustc 1.30.0 (da5f414c2 2018-10-24)

Rustc produces E0704 error code on the following (syntactically wrong) code:
```rust
mod foo {
    pub(foo) struct Bar {
        x: i32
    }
}
```
```
error[E0704]: incorrect visibility restriction
  --> src/main.rs:35:9
   |
35 |     pub(foo) struct Bar {
   |         ^^^ help: make this visible only to module `foo` with `in`: `in foo`
   |
   = help: some possible visibility restrictions are:
           `pub(crate)`: visible only on the current crate
           `pub(super)`: visible only in the current module's parent
           `pub(in path::to::module)`: visible only on the specified path

error: aborting due to previous error

For more information about this error, try `rustc --explain E0704`.
```
But but neither `rustc --explain E0704` nor `https://doc.rust-lang.org/error-index.html` is aware of this error

--- a/src/libsyntax/error_codes.rs
+++ b/src/libsyntax/error_codes.rs
@@ -363,6 +363,35 @@ and likely to change in the future.
 
 "##,
 
+E0704: r##"
+This error indicates that a incorrect visibility restriction was specified.
+
+Example of erroneous code:
+
+```compile_fail,E0704
+mod foo {
+    pub(foo) struct Bar {
+        x: i32
+    }
+}
+```
+
+To make struct `Bar` only visible in module `foo` the `in` keyword should be
+used:
+```
+mod foo {
+    pub(in crate::foo) struct Bar {
+        x: i32
+    }
+}
+# fn main() {}
+```
+
+For more information see the Rust Reference on [Visibility].
+
+[Visibility]: https://doc.rust-lang.org/reference/visibility-and-privacy.html
+"##,
+
 E0705: r##"
 A `#![feature]` attribute was declared for a feature that is stable in
 the current edition, but not in all editions.
@@ -417,6 +446,5 @@ register_diagnostics! {
     E0693, // incorrect `repr(align)` attribute format
     E0694, // an unknown tool name found in scoped attributes
     E0703, // invalid ABI
-    E0704, // incorrect visibility restriction
     E0717, // rustc_promotable without stability attribute
 }

--- a/src/test/ui/pub/pub-restricted.stderr
+++ b/src/test/ui/pub/pub-restricted.stderr
@@ -50,3 +50,4 @@ LL |         pub (in x) non_parent_invalid: usize,
 
 error: aborting due to 5 previous errors
 
+For more information about this error, try `rustc --explain E0704`.

